aten/src/ATen/native/MaxUnpooling.cpp

```diff
@@ -118,17 +175,113 @@ Tensor& max_unpooling2d_forward_out_cpu(
             output, self, indices, oheight, owidth);
       }));
   return output;
-};
+}
 
 Tensor max_unpooling2d_forward_cpu(
     const Tensor& self,
     const Tensor& indices,
-    IntArrayRef output_size) {
+    IntArrayRef output_size,
+    IntArrayRef stride,
+    IntArrayRef padding) {
   auto output = at::empty({0}, self.options());
-  max_unpooling2d_forward_out_cpu(output, self, indices, output_size);
+  max_unpooling2d_forward_out_cpu(output, self, indices, output_size, stride, padding);
+  return output;
+}
+
+static void max_unpooling1d_shape_check(
+    const Tensor& input,
+    const Tensor& gradOutput,
+    const Tensor& indices,
+    IntArrayRef output_size,
+    IntArrayRef stride,
+    IntArrayRef padding) {
+  auto owidth = output_size[0];
+  TORCH_CHECK(
+      indices.scalar_type() == at::ScalarType::Long,
+      "elements in indices should be type int64");
+  TORCH_CHECK(
+      output_size.size() == 1,
+      "There should be exactly one elements in output_size");
+  TORCH_CHECK(
+      stride.size() == 1,
+      "There should be exactly one elements in stride");
+  TORCH_CHECK(
+      padding.size() == 1,
+      "There should be exactly one elements in padding");
+  TORCH_CHECK(
+      (input.ndimension() == 3),
+      "Input to max_unpooling1d should be a 3d Tensor");
+  TORCH_CHECK(
+      input.sizes() == indices.sizes(),
+      "Shape of indices should match shape of input");
+
+  TORCH_CHECK(input.numel() > 0, "Input must be non-empty");
+  TORCH_CHECK(
+      stride[0] > 0,
+      "strides should be greater than zero, but got stride: ",
+      stride);
+
+  int dimw = 2;
+
+  if (gradOutput.defined()) {
+    if (owidth != gradOutput.size(dimw)) {
+      AT_ERROR(
+          "Inconsistent gradOutput size. oW= ",
+          owidth,
+          ". gradOutput: ",
+          gradOutput.size(dimw));
+    }
+    TORCH_CHECK(
+        gradOutput.ndimension() == input.ndimension(),
+        "gradOutput and input Tensors should have same number of dimensions");
+  }
+}
+
+Tensor& max_unpooling1d_forward_out_cpu(
+    Tensor& output,
+    const Tensor& self_,
+    const Tensor& indices_,
+    IntArrayRef output_size,
+    IntArrayRef stride,
+    IntArrayRef padding) {
+  auto oheight = output_size[0];
+  auto owidth = 1;
+  TORCH_CHECK(output.is_contiguous(), "output must be contiguous");
+  max_unpooling1d_shape_check(
+      self_, Tensor(), indices_, output_size, stride, padding);
+  /* extend one dimension for tensor so that it can use max_unpooling2d_backward_out_cpu_frame method. */
+  auto self = self_.contiguous().unsqueeze(3);
```

Comment:
The input could be 2D or 3D (batch), so you need to condition the unsqueeze on that. The original code actually has the same issue and does not handle non-batch input correctly. See here https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L694

Could you change both this and the code in functional.py to do the unsqueeze conditionally on whether the input is 2D or 3D?